src/interactivity/base/InteractivityFactory.hpp

```diff
@@ -3,27 +3,27 @@
 
 #pragma once
 
-#include "precomp.h"
-
-#include "ApiDetector.hpp"
-
-#include "../inc/IInteractivityFactory.hpp"
-
-#include <map>
-
-#pragma hdrstop
-
 namespace Microsoft::Console::Interactivity
 {
-    class InteractivityFactory final : public IInteractivityFactory
+    namespace Win32
+    {
+        class WindowDpiApi;
+    }
+
+    class IInputServices;
+    class ISystemConfigurationProvider;
+    class IConsoleInputThread;
+    class IConsoleControl;
+
+    class InteractivityFactory
     {
     public:
         [[nodiscard]] NTSTATUS CreateConsoleControl(_Inout_ std::unique_ptr<IConsoleControl>& control);
         [[nodiscard]] NTSTATUS CreateConsoleInputThread(_Inout_ std::unique_ptr<IConsoleInputThread>& thread);
 
-        [[nodiscard]] NTSTATUS CreateHighDpiApi(_Inout_ std::unique_ptr<IHighDpiApi>& api);
+        [[nodiscard]] NTSTATUS CreateHighDpiApi(_Inout_ std::unique_ptr<Win32::WindowDpiApi>& api);

```

Comment:
Only thing I'm worried about with these is that there's a "OneCore" version of all of these, last I checked, that might not be showing up in the msbuild chain compilation and will only show up with the Windows Razzle compilation chain. (The files are here in the open, but they don't have a .vcxproj attached).

Reply:
And uh... we could technically set a compilation flag that compiled EITHER the win32 ones OR the onecore ones depending on the OS SKU if you wanted to remove the virtual that way.

Reply:
I've reverted all OneCore changes.